### PR TITLE
Bug/DES-1144 - Fix user directive to support other project authors (hotfix)

### PR DESCRIPTION
**Pull request by Strmiska**
Created a condition inside the `dsUserList` directive to accept a list of user objects for other
Added try/except catch if user's name is not found in database

### CODE REVIEW

File: designsafe/apps/api/agave/filemanager/public_search_index.py [hunk 1]
```diff
@@ -144,9 +144,13 @@ def to_file(self):
             dict_obj['meta']['piLabel'] = '{last_name}, {first_name}'.format(
                 last_name=pi_user['last_name'], first_name=pi_user['first_name'])
         else:
-            pi_username = get_user_model().objects.get(username=self.project['value']['pi'])
-            dict_obj['meta']['piLabel'] = '{last_name}, {first_name}'.format(
-                last_name=pi_username.last_name, first_name=pi_username.first_name)
+            try:
+                pi_username = get_user_model().objects.get(username=self.project['value']['pi'])
```
Comment (rstijerina) [inline designsafe/apps/api/agave/filemanager/public_search_index.py]: if you have the `pi` object defined above on line 140, why not use it for all instances of `self.project['value']['pi']`?

Also the `pi_username` object is really a User instance, not a username, so `pi_username` might be `pi_user` instead

File: designsafe/static/scripts/ng-designsafe/directives/ng-designsafe-directives.js [hunk 2]
```diff
@@ -277,30 +277,60 @@ export function dsUserList(UserService) {
...
+            });
+            element.text(formattedNames);
           });
-          element.text(formattedNames);
-        });
+
+        } else {
+          UserService.getPublic(userReq).then((res) => {
+            let userData = res.userData;
+            let formattedNames = '';
```
Comment (rstijerina) [inline designsafe/static/scripts/ng-designsafe/directives/ng-designsafe-directives.js]: some of this is repeated above - should it be outside the if statement?

File: designsafe/apps/api/agave/filemanager/public_search_index.py (hunk 1 above)
Comment (Strmiska) [reply designsafe/apps/api/agave/filemanager/public_search_index.py]: Good call. I missed the other `pi` references there, and on the `pi_username` yes that makes more sense I'll change that to `pi_user`.

File: designsafe/static/scripts/ng-designsafe/directives/ng-designsafe-directives.js (hunk 2 above)
Comment (Strmiska) [reply designsafe/static/scripts/ng-designsafe/directives/ng-designsafe-directives.js]: Yep! I'll move it out.

**Pull request merged**